### ISSUE TITLE
HMAC without an explicit digestmod, which breaks on 3.8

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#9800]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#9800|
|Type|enhancement|
|Created|2020-04-12 18:47:28Z|

The HMAC.__init__() method's digestmod parameter became required
in Python 3.8:

https://github.com/python/cpython/commit/8bb0b5b03cffa2a2e74f248ef479a9e7fbe95cf4
https://github.com/python/cpython/commit/f33c57d5c780da1500619f548585792bb5b750ee

This is the same problem which was mentioned in [#9733](https://github.com/twisted/twisted/issues/9733)

<details><summary>Searchable metadata</summary>

```
trac-id__9800 9800
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_8_Support Python 3.8 Support
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1586717248083685 1586717248083685
changetime__1586745730984349 1586745730984349
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="altendky's avatar" src="https://avatars.githubusercontent.com/u/543719?s=50" width="50" height="50">](https://github.com/altendky)<a name="note_2"></a>|@altendky commented|
|-|-|

Approved: https://github.com/twisted/twisted/pull/1251#pullrequestreview-391923357

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @rodrigc|
|-|-|

Approved: https://github.com/twisted/twisted/pull/1251#pullrequestreview-391923357

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_5"></a>|@rodrigc set status to `closed`|
|-|-|

In changeset f58c702dac599695bbe9a3e047f8550e11274cdc
```
#!CommitTicketReference repository="" revision="f58c702dac599695bbe9a3e047f8550e11274cdc"
Merge pull request #1251 from twisted/9800-rodrigc-hmac

Add digestmod parameter to HMAC.__init__() to fix Python 3.8+

Author: rodrigc
Reviewers: altendky 
Fixes: ticket:9800
```
